components/server/ee/src/bitbucket/bitbucket-app-support.ts

```diff
@@ -41,44 +42,56 @@ export class BitbucketAppSupport {
         const workspaces =
             (await api.workspaces.getWorkspaces({ pagelen: 100 })).data.values?.map((w) => w.slug!) || [];
 
-        const reposPromise = Promise.all(
-            workspaces.map((workspace) =>
-                api.repositories
+        const fetchAllRepos = async (workspace: string) => {
+            const reposResponse: (Schema.Repository[] | undefined)[] = [];
+            let page = "1";
+            let hasMorePages = true;
+            const pagelen = 100;
+
+            while (hasMorePages) {
+                const response = await api.repositories
                     .list({
                         workspace,
-                        pagelen: 100,
+                        pagelen,
+                        page,
                         role: "admin", // installation of webhooks is allowed for admins only
                     })
                     .catch((e) => {
                         console.error(e);
```

Comment:
Is there a simple way to show errors like this to users without requiring user interaction like alert() or confirm() do? I like the pattern in GitHub where it adds a blue banner at the top with error or success messages.

Reply:
Do you have a screenshot of an example? Either on our own platform or on GitHub.

Reply:
That's a nice idea. I think we were discussing this already and the toast notification was included as well. 
cc. @gtsiolis

Reply:
I have opened separate issue #8933.
Keen to ship this fix.